### ISSUE TITLE
[Question] Meaning of lightcone_quantities={'velocity'}

### ISSUE THREAD

**Issue by jjsienna** [21cmfast/21cmFAST#167]
Hello,

I've recently started using 21cmFAST and have a beginner question.
What I would like to obtain is the line-of-sight and transverse velocities in the lightcone.
Following the [lightcone tutorial](https://21cmfast.readthedocs.io/en/latest/tutorials/lightcones.html), I have used:

```
lightcone = p21c.run_lightcone(
    redshift = 7.0,
    max_redshift = 12.0,
    user_params = {"HII_DIM":150, "BOX_LEN": 600},
    lightcone_quantities=("brightness_temp", 'density', 'velocity'),
    global_quantities=("brightness_temp", 'density', 'xH_box', 'velocity'),
    direc='_cache'
)
```

This outputs a lightcone object but I'm confused as to what the 3D array `lightcone.velocity[:,:,k ]` is representing. Is this the LOS velocity at the redshift index k or is it just the magnitude ? If it is not the LOS velocity, I believe I can extract that information  by passing `'lowres_vx', 'lowres_vy', 'lowres_vz'` to `lightcone_quantities={}`, but would it be possible for someone to clarify which direction x,y,z is taken as the redshift direction (if the lightcones are generated by simply stacking 
 the boxes in one direction)?
 
Thank you

**Comment by BradGreig** [comment]
Hi @jjsienna,

At the moment, it is not possible to obtain the lightcone of both the transverse and line-of-sight velocities. Currently, `lightcone.velocity` returns the light-cone of the line-of-sight velocity, which is fixed to be the z-direction.

Adding `lowres_vx` etc. to `lightcone_quantities={}` will not give you what you are after. `lowres_vx' corresponds to the z=0 velocity field, and thus cannot be used to create a lightcone (as it is not redshift dependent).

That being said, there is no reason why you shouldn't be able to obtain each component of the velocities. Or even to generate light-cones along any direction (not just fixed to the z-direction). Probably I was trying to avoid excessive memory usage and extra complexity by fixing both.

I'll raise it as a feature request, but it might take me (or someone else) a bit of time to get to it. However, in the interim there are a couple of options:

1) Make your own branch and try and implement the minimum requirements that you need for yourself. Adding the x- and y- components to `PerturbField` should be trivial, which would provide you what you need. Its being clever about selecting 1/3 velocity components (and which direction) to be flexible and avoid unnecessary excess memory usage that'll take a while

2) I can create the branch that deals with 1) for you

Let me know if you want to give it a go yourself, or have me create a temporary branch for now.

**Comment by jjsienna** [comment]
Hello @BradGreig 

Thank you very much for the quick reply and the confirmation on the direction of the lightcone and what `lightcone.velocity` is representing. Right after I posted, I realized that the parallel plane approximation will probably work for the problem that I have in hand, so I think having the z-direction velocity is sufficient for now. However, I know that in the future when we have to simulate larger sky area, the parallel approximation will fail. When that time comes, I will revisit and give (1) a go myself. Sorry for causing random noise.
